test/test_proxy_tensor.py

```diff
@@ -1403,6 +1404,56 @@ def forward(self, a_1):
     div = torch.ops.aten.div.Tensor(a_1, pow_1);  a_1 = pow_1 = None
     return div""")
 
+    def test_make_fx_with_custom_tracer_preserving_nn_module_stack(self):
+
+        class Bar(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                return x + 1
+
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.bar = Bar()
+
+            def forward(self, x):
+                return x + self.bar(x)
+
+        gm = make_fx(Foo())(torch.randn(4, 4))
+        for node in gm.graph.nodes:
+            self.assertTrue("nn_module_stack" not in node.meta)
+
+        foo = Foo()
+
+        def functional_call(*args, **kwargs):
+            with stateless._reparametrize_module(foo, {}):
+                return foo(*args, **kwargs)
+
+        functional_call._orig_mod = foo
+
+        gm_with_stack = make_fx(functional_call)(torch.randn(4, 4))
+        found = False
+        for node in gm_with_stack.graph.nodes:
+            if "nn_module_stack" in node.meta:
+                if len(node.meta["nn_module_stack"]) == 1:
+                    self.assertExpectedInline(
+                        str(node.meta["nn_module_stack"]),
+                        """OrderedDict([('', ('', <class '__main__.TestSymbolicTracing.test_make_fx_with_custom_tracer_preserving_nn_module_stack.<locals>.Foo'>))])"""  # noqa: B950
+                    )
+                    found = True
+                elif len(node.meta["nn_module_stack"]) == 2:
+                    self.assertExpectedInline(
+                        str(node.meta["nn_module_stack"]),
+                        """OrderedDict([('', ('', <class '__main__.TestSymbolicTracing.test_make_fx_with_custom_tracer_preserving_nn_module_stack.<locals>.Foo'>)), ('bar', ('bar', <class 'torch.fx.experimental.proxy_tensor.Bar'>))])"""  # noqa: B950
```

Comment:
Why do we end up with `torch.fx.experimental.proxy_tensor.Bar` as the full path, if `Bar` was created inside of this test 🤔

Reply:
yeah this seems wrong lol